src/Documentation/doc/en/programming/hackingalgorithms.md

```diff
@@ -36,96 +50,134 @@ The general logic goes like this:
 
-**Difficulty**: Hard
+Don't forget to explore the [NS API](github.com/bitburner-official/bitburner-src/blob/stable/markdown/bitburner.ns.md) for other functions that might be useful in these calculations!
+
+## Shotgun Batchers
+
+**Difficulty**: Medium
 
 Pros:
 
-- Maximum potential income
+- Easy to implement with a working controller script
+- Much better income than EHT
 
 Cons:
 
-- Very difficult to implement without prior programming knowledge
-- Very difficult to make work on servers with less than 1TB of RAM
+- Requires a dedicated controller script
+- Can be real-life hardware heavy
 
-Batch algorithms utilize a master script that uses `exec` many scripts which utilize a relevant hacking function in batches.
+A shotgun batcher implements uses a controller script to launch as many batches in parallel as possible.
+Each function has a delay to make them all the same run time, meaning if they are launched in the correct order they will finish in the correct order.
+Very little - potentially 0 - time is needed between batches if they are launched correctly.
 
-The scripts used to execute the hacking functions are even simpler than the previous algorithms but a complex controller is required to calculate the effect, time taken, and the necessary delay.
+In other words, instead of launching batches sequentially like this:
 
-    await ns.sleep(a bit)
-    await ns.hack(target) // or grow, or weaken
+    |=Batch=||=Batch=||=Batch=||=Batch=||=Batch=||=Batch=|
 
-A few things need to be known before this algorithm can be implemented:
+A shotgun batcher uses the available RAM in parallel to hit the target multiple times almost instantly - hence the name.
 
-- The effects of hack and grow depend on the server security level, a higher security level results in a reduced effect.
-  You only want these effects to occur when the security level is minimized.
-- The time taken to execute hack, grow, or weaken is determined when the function is called and is based on the security level of the target server and your hacking level.
-  You only want these effects to start when the security level is minimized.
-- The effects of hack, grow, and weaken, are determined when the time is completed, rather than at the beginning.
-  Hack should finish when security is minimum and money is maximum.
-  Grow should finish when security is minimum, shortly after a hack occurred.
-  Weaken should occur when security is not at a minimum due to a hack or grow increasing it.
+    |=Batch=|
+    |=Batch=|
+    |=Batch=|
+    |=Batch=|
+    |=Batch=|
+    |=Batch=|
+
+Launching potentially thousands of batches at the same time is intensive on real-life hardware. It's recommended to limit the number of parallel batches to around 100 000 to reduce the risk of the game crashing.
```

Comment:
Added a little detail here, since the steam app doesn't crash, i.e. it's slightly less scary than it could be.
```suggestion
Launching potentially thousands of batches at the same time is intensive on real-life hardware. It's recommended to limit the number of parallel batches to around 100 000 to reduce the risk of the game soft-crashing. (Also called a "black screen" because of how it appears when this happens.)
```